pkg/identityserver/client_registry.go

```diff
@@ -154,11 +154,20 @@ func (is *IdentityServer) createClient( //nolint:gocyclo
 		return nil, err
 	}
 
+	var apikey string
+	if authInfo, err := is.authInfo(ctx); err == nil {
+		apikey = authInfo.GetApiKey().GetApiKey().GetId()
```

Comment:
In which cases would the error be not `nil` and we would still like to continue ?

Reply:
Since it wasn't being validated before I initially ignored error related to the `authInfo`. Reflecting on it, the authInfo being unavailable should trigger an error on the client creation so I just return the error if there is one.